src/main/java/com/iheartradio/m3u8/M3uParser.java

```diff
@@ -14,17 +14,28 @@
         mEncoding = encoding;
     }
 
-    Playlist parse() throws ParseException {
+    Playlist parse(ParsingMode parsingMode) throws ParseException {
         final ParseState state = new ParseState(mEncoding);
         final TrackHandler trackHandler = new TrackHandler();
 
         try {
             state.setMedia();
-
+            
             while (mScanner.hasNext()) {
                 final String line = mScanner.next();
-                validateLine(line);
-
+                
+                try {
+                    validateLine(line);
+                } catch (ParseException parseException) {
+                    switch(parsingMode) {
+                        case STRICT:
+                            throw parseException;
+                        case LENIENT:
+                        default:
+                            state.unknownTags.add(line);
```

Comment:
The `M3uParser` parses non-extended playlists. This means there are no tags (or tags are read as comments) so lenient and strict mode have no meaning.